pkg/reconciler/apiserversource/apiserversource_test.go

```diff
@@ -368,9 +367,9 @@ func TestReconcile(t *testing.T) {
 	}
 
 	defer logtesting.ClearAll()
-	table.Test(t, MakeInjectionFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
+	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
 		r := &Reconciler{
-			Base:                  reconciler.NewInjectionBase(ctx, controllerAgentName, cmw),
+			Base:                  reconciler.NewBase(ctx, controllerAgentName, cmw),
```

Comment:
Just curious on our use for Context (which seems to be for passing in options). It's not against this PR but just wanted to clarify about it, so this seemed just as good of a place as any. From the golang page;
https://golang.org/pkg/context/

So, just curious if we're doing something that's actually not what we should be doing.
@mattmoor

Reply:
we are stretching the usage of context a bit, true. The thread that ties this together is that we are spun off the main thread? And all other threads should get the shared informer caches.

Reply:
Yeah, it's a stretch.

Effectively what we're banking on here is that there are two functions with type `context.Context -> informers.MyKindInformer`, one for realz and one for testing.  By linking the package to access these functions, we "register" them (and their transitive deps), so any properly initialized context has them.